kv/kvserver: TestReplicateQueueExpirationLeasesOnly failed

kv/kvserver.TestReplicateQueueExpirationLeasesOnly [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9143037?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9143037?buildTab=artifacts#/) on master @ [de5b08a418390a803357ba4b74a55a4eec26e16b](https://github.com/cockroachdb/cockroach/commits/de5b08a418390a803357ba4b74a55a4eec26e16b):


```
=== RUN   TestReplicateQueueExpirationLeasesOnly
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueExpirationLeasesOnly171761207
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:2414: initial: epochLeases=315 expLeases=2
    replicate_queue_test.go:2421: enabling: epochLeases=297 expLeases=20
    replicate_queue_test.go:2421: enabling: epochLeases=274 expLeases=43
    replicate_queue_test.go:2421: enabling: epochLeases=255 expLeases=62
    replicate_queue_test.go:2421: enabling: epochLeases=234 expLeases=83
    replicate_queue_test.go:2421: enabling: epochLeases=214 expLeases=89
    replicate_queue_test.go:2421: enabling: epochLeases=195 expLeases=91
    replicate_queue_test.go:2421: enabling: epochLeases=180 expLeases=87
    replicate_queue_test.go:2421: enabling: epochLeases=166 expLeases=78
    replicate_queue_test.go:2421: enabling: epochLeases=155 expLeases=71
    replicate_queue_test.go:2421: enabling: epochLeases=143 expLeases=70
    replicate_queue_test.go:2421: enabling: epochLeases=130 expLeases=66
    replicate_queue_test.go:2421: enabling: epochLeases=119 expLeases=66
    replicate_queue_test.go:2421: enabling: epochLeases=110 expLeases=66
    replicate_queue_test.go:2421: enabling: epochLeases=103 expLeases=65
    replicate_queue_test.go:2421: enabling: epochLeases=93 expLeases=64
    replicate_queue_test.go:2421: enabling: epochLeases=85 expLeases=57
    replicate_queue_test.go:2421: enabling: epochLeases=76 expLeases=55
    replicate_queue_test.go:2421: enabling: epochLeases=68 expLeases=53
    replicate_queue_test.go:2421: enabling: epochLeases=60 expLeases=57
    replicate_queue_test.go:2421: enabling: epochLeases=49 expLeases=57
    replicate_queue_test.go:2421: enabling: epochLeases=39 expLeases=62
    replicate_queue_test.go:2421: enabling: epochLeases=29 expLeases=64
    replicate_queue_test.go:2421: enabling: epochLeases=21 expLeases=67
    replicate_queue_test.go:2421: enabling: epochLeases=11 expLeases=66
    replicate_queue_test.go:2419: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/pkg/kv/kvserver/replicate_queue_test.go:2419
        	Error:      	Condition never satisfied
        	Test:       	TestReplicateQueueExpirationLeasesOnly
    replicate_queue_test.go:2421: enabling: epochLeases=2 expLeases=67
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueExpirationLeasesOnly171761207
--- FAIL: TestReplicateQueueExpirationLeasesOnly (214.69s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueExpirationLeasesOnly.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25661

kv/kvserver.TestReplicateQueueExpirationLeasesOnly [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9163527?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9163527?buildTab=artifacts#/) on master @ [b89fa2cc4bc1fb9447ab1009b34b6f354f9618f0](https://github.com/cockroachdb/cockroach/commits/b89fa2cc4bc1fb9447ab1009b34b6f354f9618f0):


```
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_pipeliner.go:291
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnSeqNumAllocator).SendLocked
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_seq_num_allocator.go:105
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*txnHeartbeater).SendLocked
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_interceptor_heartbeater.go:246
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord.(*TxnCoordSender).Send
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvclient/kvcoord/txn_coord_sender.go:525
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:1001
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Send
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/txn.go:1060
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:831
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/txn.go:675
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).CommitInBatch
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/txn.go:723
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*CrossRangeTxnWrapperSender).Send.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:236
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:965
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/txn.go:928
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:964
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).TxnWithAdmissionControl
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:927
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:902
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*CrossRangeTxnWrapperSender).Send
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:228
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).sendUsingSender
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:1001
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).send
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:984
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:831
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Run
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:854
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).scan
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:475
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Scan
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/db.go:486
        	            	Wraps: (2) aborted during DistSender.Send
        	            	Wraps: (3) context deadline exceeded
        	            	Error types: (1) *errbase.opaqueWrapper (2) *errutil.withPrefix (3) context.deadlineExceededError
        	Test:       	TestReplicateQueueExpirationLeasesOnly
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/33e1d369c27b9c01b2b6009c561815a3/logTestReplicateQueueExpirationLeasesOnly2688684404
--- FAIL: TestReplicateQueueExpirationLeasesOnly (130.83s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueExpirationLeasesOnly.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -2363,6 +2363,7 @@ func TestReplicateQueueExpirationLeasesOnly(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderRace(t) // too slow under stressrace
+	skip.UnderDeadlock(t)
 	skip.UnderShort(t)
 
 	ctx := context.Background()

